agent/taskresource/logrouter/logconfig_unix.go

```diff
@@ -0,0 +1,144 @@
+// +build linux
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package logrouter
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	generator "github.com/awslabs/go-config-generator-for-fluentd-and-fluentbit"
+)
+
+const (
+	// LogRouterTypeFluentd is the type of a fluentd log router.
+	LogRouterTypeFluentd = "fluentd"
+
+	// LogRouterTypeFluentbit is the type of a fluentbit log router.
+	LogRouterTypeFluentbit = "fluentbit"
+
+	// socketInputNameFluentd is the name of the socket input plugin for fluentd.
+	socketInputNameFluentd = "unix"
+
+	// socketInputNameFluentbit is the name of the socket input plugin for fluentbit.
+	socketInputNameFluentbit = "forward"
+
+	// socketInputPathOptionFluentd is the key for specifying socket path for fluentd.
+	socketInputPathOptionFluentd = "path"
+
+	// socketInputPathOptionFluentbit is the key for specifying socket path for fluentbit.
+	socketInputPathOptionFluentbit = "unix_path"
+
+	// outputLogOptionKeyFluentd is the key for the log option that specifies output plugin name for fluentd.
+	outputLogOptionKeyFluentd = "@type"
+
+	// outputLogOptionKeyFluentbit is the key for the log option that specifies output plugin name for fluentbit.
+	outputLogOptionKeyFluentbit = "Name"
+
+	// includePatternKey is the key for include pattern.
+	includePatternKey = "include-pattern"
+
+	// excludePatternKey is the key for exclude pattern.
+	excludePatternKey = "exclude-pattern"
+
+	// socketPathFormat is the format for socket file path.
+	socketPathFormat = "/var/run/%s.sock"
+)
+
+// generateConfig generates a FluentConfig object that contains all necessary information to construct
+// a fluentd or fluentbit config file for the log router.
+func (logRouter *LogRouterResource) generateConfig() (generator.FluentConfig, error) {
+	config := generator.New()
+
+	// Specify log stream input, which is a unix socket that will be used for communication between log router container
+	// and other containers.
+	var inputName, inputPathOption string
+	if logRouter.logRouterType == LogRouterTypeFluentd {
+		inputName = socketInputNameFluentd
+		inputPathOption = socketInputPathOptionFluentd
+	} else {
+		inputName = socketInputNameFluentbit
+		inputPathOption = socketInputPathOptionFluentbit
+	}
+	config.AddInput(inputName, "", map[string]string{
+		inputPathOption: fmt.Sprintf(socketPathFormat, logRouter.logRouterType),
```

Comment:
This is fine, but the unix socket path could be the same in the container for both. That will make it easier on the host path/docker connection side of things too.

So I'd change your socketPathFormat to just a constant, with the value `/var/run/fluent.sock`.

Reply:
yeah makes sense, updated to /var/run/fluent.sock